### ISSUE TITLE
💅 noUnusedVariables still checking function arguments

### ISSUE THREAD

**Issue by lionskape** [biomejs/biome#4554]
### Environment information

```bash
There is a doc: https://biomejs.dev/linter/rules/no-unused-variables/
It said: "From v1.9.0, the rule won’t check unused function parameters any more. If you want to report unused function parameters, enable noUnusedFunctionParameters."

Today we have a version 1.9.4, so the issue still exists (take a look at playground)
```

### Rule name

lint/correctness/noUnusedVariables

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=YwBvAG4AcwB0ACAAdABlAHMAdAAgAD0AIAAoAHYAMQA6ACAAdQBuAGsAbgBvAHcAbgAsACAAdgAyADoAIABzAHQAcgBpAG4AZwApADoAIABzAHQAcgBpAG4AZwAgAD0APgAgAHsACgAgACAAIAAgAHIAZQB0AHUAcgBuACAAdgAyADsACgB9ADsACgAKAHQAZQBzAHQAKAAwACwAIAAnADAAJwApAA%3D%3D

### Expected result

No issues on that playground

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by lionskape** [comment]
Oh, so sad. So, I'm looking forward to the second version

**Comment by cparrosteamworks** [comment]
Looking forward to a fix here 😄

**Comment by sparebytes** [comment]
I hit this issue with Biome v1.9.4.

Playground Link: https://biomejs.dev/playground/?lintRules=all&code=ZgB1AG4AYwB0AGkAbwBuACAAdABlAHMAdAAoAHUAbgB1AHMAZQBkADoAIABzAHQAcgBpAG4AZwApADoAIABzAHQAcgBpAG4AZwAgAHsACgAgACAAIAAgAHIAZQB0AHUAcgBuACAAIgBoAGUAbABsAG8AIgA7AAoAfQA7AAoACgB0AGUAcwB0ACgAIgAiACkACgA%3D
